app/grandchallenge/core/db/postgres_iam/base.py

```diff
@@ -1,17 +1,72 @@
+import functools
+from datetime import timedelta
+
 import boto3
 from django.db.backends.postgresql import base
+from django.utils import timezone
+
+
+def thread_local_cache(ttl, max_size=128):
+    def decorator(func):
+        function_cache = {}
```

Comment:
Doesn't this create a cache that all threads can access, rather than a thread-local cache? I'm probably misunderstanding what you mean by thread-local...

Reply:
Okay, I meant just local to the worker running the code, not shared across workers as done with the django cache. I have replaced it with a third party package for simplicity.